Api/Client.php

```diff
@@ -310,6 +307,75 @@ private function removeLastLineFromFiles($files = null, $noOfLinesToRemove = 1)
 
     /**
      *
+     * @param array $lines
+     * @param int   $noOfLinesToRemove
+     *
+     * @return array
+     */
+    private function removeLastLines(array $lines, $noOfLinesToRemove = 1)
+    {
+        $removeFrom = count($lines) - $noOfLinesToRemove;
+
+        for( $i = $removeFrom; $i < count($lines); $i++ ) {
```

Comment:
This count will be less each time, you need to do this first before starting the for loop. 

We should really start writing tests, you will catch this in your development process ;)